dashboard/dashboard.py

```diff
@@ -0,0 +1,777 @@
+        'count', ascending=False).head(12)
+    pie_chart = alt.Chart(aggregated_data).mark_arc().encode(
+        theta=alt.Theta(field='count', type='quantitative',
+                        title='Count').stack(True),
+        color=colour,
+        tooltip=[field, 'count'],
+        # order=alt.Order('count', sort='descending')
+    ).properties(
+        title=f"{name} Distribution", width=500, height=400).mark_arc(outerRadius=130)
+
+    return pie_chart
+
+
+def plot_filter_pie_tags(df: pd.DataFrame, selected_filter: list[str], field: str, tab: str,
+                         name: str):
+    """
+    Pie chart that displays the distribution of tags based on either the judges, courts or verdicts
+    """
+    filtered_data = df[df[tab].isin(selected_filter)]
+    aggregated_data = filtered_data.groupby(field).sum().reset_index()
+    aggregated_data = aggregated_data.sort_values(
+        'count', ascending=False).head(12)
+    pie_chart = alt.Chart(aggregated_data).mark_arc().encode(
+        alt.Theta('count', type='quantitative', title='Tag Count'),
+        color=alt.Color(field=field, type='nominal', title=name),
+        tooltip=[field, 'count']
+    ).properties(width=450, height=400).mark_arc(outerRadius=135)
+    return pie_chart
+
+
+def plot_cases_over_months(df: pd.DataFrame):
+    """Plot of all the case count, when aggregated by month"""
+    title = alt.TitleParams(
+        'Cases Heard, Aggregated per Month', anchor='start')
+    return alt.Chart(df.reset_index(), title=title).mark_bar().encode(
+        x=alt.X('month', title='Month', sort=None,
+                axis=alt.Axis(labelAngle=0)),
+        y=alt.Y('case_count', title='Case Count')).properties(width=450, height=400)
+
+
+def draw_line(data:pd.DataFrame, field: str, filter_title: str):
+    """Graph with a singular line: field cumulative count vs time"""
+    data.loc[:, 'overall_sum'] = data['case_count'].cumsum()
+    return alt.Chart(data).mark_line(opacity=1, thickness=0.01).encode(
+        x=alt.X('court_date:T', title='Date of the Case'),
+        y=alt.Y('overall_sum:Q', title='Case Count'),
+        color=alt.Color(field, title=filter_title.title(),
+                        legend=alt.Legend(orient="top")),
+        tooltip=[field, 'overall_sum']
+    )
+
+
+def select_filter(df: pd.DataFrame, field: str, filter_title: str):
+    """Function that groups different databases into one line chart, one per field selected"""
+    graph = []
+    for i in set(df[field]):
+        graph.append(draw_line(df[df[field] == i], field, filter_title))
+    return alt.layer(*graph).properties(
+        title=f"""Cases for Selected {filter_title.title()}s, Over Time""")
+
+
+def subscribe_to_court(courts: list):
+    """Allowing a SNS subscription to specific courts"""
+    st.header("Subscribe to Notifications")
+
+    email_input = st.text_input("Enter your email to subscribe:")
+    email = re.search(r"[\w_.-]+@[\w_.-]+[.]+[\w]+", email_input)
```

Comment:
Definitely add the more robust regex that is on stack overflow to catch all the emails.